src/Compilers/CSharp/Portable/Symbols/MethodSymbol.cs

```diff
@@ -1069,6 +1069,11 @@ public override bool Equals(Symbol other, TypeCompareKind compareKind)
                 return sms.Equals(this, compareKind);
             }
 
+            if (other is NativeIntegerMethodSymbol nms)
```

Comment:
>if [](start = 12, length = 2)

consider `return other switch { SubstitutedMethodSymbol sms => sms.Equals(this, compareKind), NativeIntegerMethodSymbol nms => nms.Equals(this, compareKind), _ => base.Equals(other, compareKind) };`